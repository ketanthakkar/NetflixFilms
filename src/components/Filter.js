import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { sortMovies, fetchMovies } from '../actions/index';

export class Filter extends Component {
    handleSortClick = (event) => {
      const sortBy = event.target.id === 'release' ? 'release_date' : 'vote_average';
      this.props.sortMovies(sortBy);
      this.props.fetchMovies();
    };

    render() {
      const { movieCount, activeSorting } = this.props;

      return (
            <section className="filter-container">
                <span className="results">{movieCount} movies found</span>
                <div className="sort-detail">
                    <span>Sort by</span>
                    <span id="release" 
                          className={activeSorting === 'release_date' ? 'highlight-color' : ''} 
                          onClick={this.handleSortClick}>
                      release date
                    </span>
                    <span id="rating" 
                          className={activeSorting === 'vote_average' ? 'highlight-color' : ''} 
                          onClick={this.handleSortClick}>
                      rating
                    </span>
                </div>
            </section>
      );
    }
}

Filter.propTypes = {
  movieCount: PropTypes.number,
  activeSorting: PropTypes.string,
  sortMovies: PropTypes.func,
  fetchMovies: PropTypes.func
};

const mapStateToProps = state => ({
  movieCount: state.movies.movieData.length,
  activeSorting: state.sortBy.sortBy,
});

const mapDispatchToProps = {
  sortMovies,
  fetchMovies,
};

export default connect(mapStateToProps, mapDispatchToProps)(Filter);
