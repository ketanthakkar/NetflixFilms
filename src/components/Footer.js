import React from 'react';
import styled from 'styled-components';

const FooterSection = styled.footer`
    background-color: rgba(0, 0, 0, 0.9);
    padding: 10px;
    width: 100%;
`;

const NetflixText = styled.h4`
    color: #F65262;
    width: 90%;
    display: block;
    margin: 0 auto;
`;

const Footer = () => (
    <FooterSection>
      <NetflixText>netflixroulette</NetflixText>
    </FooterSection>
);

export default Footer;
