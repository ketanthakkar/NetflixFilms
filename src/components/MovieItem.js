import React from 'react';
import PropTypes from 'prop-types';
import { Link } from "react-router-dom";

const MovieItem = ({movieItem}) => {
    return (
        <section className="movieitem-container">
            <figure>
                <Link to={`/film/${movieItem.id}`}>
                    <img className="movieitem-img" src={movieItem.poster_path} alt={movieItem.tagline}/>
                </Link>
                <figcaption className="movieitem-detail">
                    <span>{movieItem.title.toUpperCase()}</span>
                    <span className="movieitem-year">{movieItem.release_date}</span>
                </figcaption>
                <span className="movieitem-genres">{movieItem.genres.join(" & ")}</span>
            </figure>
        </section>
    )
}

MovieItem.propTypes = {
    movieItem: PropTypes.object
};

export default MovieItem;