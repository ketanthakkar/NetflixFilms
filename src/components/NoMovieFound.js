import React from 'react';
import PropTypes from 'prop-types';

const NoMovieFound = () => (
  <div className="nofound-content">
    <h2 className="nofound-text">No films found</h2>
  </div>
);

NoMovieFound.propTypes = {
  message: PropTypes.string
};

export default NoMovieFound;