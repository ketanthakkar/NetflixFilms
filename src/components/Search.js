import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchMovies, filterMovies, filterMoviesBy } from '../actions/index';

export class Search extends Component {
    handleSearchClick = (event) => {
      const filterBy = event.target.id === 'genre-btn' ? 'genres' : 'title';
      this.props.filterMoviesBy(filterBy);
    }

    handleChange = (event) => {
      this.props.filterMovies(event.target.value);
    }

    render() {
      const { activeSearchBy, searchStr } = this.props;

      return (
            <section className="search-section">
                <h4 className="white-text">FIND YOUR MOVIE</h4>
                <input type="text" placeholder="Enter title/genre" className="white-text search-title"
                        onChange={this.handleChange}></input>
                <div className="search-container">
                    <div className="search-selection">
                        <label className="white-text">SEARCH BY</label>
                        <button id="title-btn" 
                                className={activeSearchBy === 'title' ? 'white-text' : 'white-text nonselected-color'} 
                                onClick={this.handleSearchClick}>
                          TITLE
                        </button>
                        <button id="genre-btn" 
                                className={activeSearchBy === 'genres' ? 'white-text' : 'white-text nonselected-color'} 
                                onClick={this.handleSearchClick}>
                          GENRE
                        </button>
                    </div>
                    <Link to={`/search/${encodeURIComponent(searchStr)}`}>
                        <button className="white-text" onClick={this.props.fetchMovies}>SEARCH</button>
                    </Link>
                </div>
            </section>
      );
    }
}

Search.propTypes = {
  filterMovies: PropTypes.func,
  fetchMovies: PropTypes.func,
  filterMoviesBy: PropTypes.func,
  activeSearchBy: PropTypes.string,
  searchStr: PropTypes.string,
};

const mapStateToProps = state => ({
  activeSearchBy: state.search.searchBy,
  searchStr: state.search.searchStr,
});

Search.defaultProps = {
  onSearch: () => { },
};

const mapDispatchToProps = {
  fetchMovies,
  filterMovies,
  filterMoviesBy,
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
