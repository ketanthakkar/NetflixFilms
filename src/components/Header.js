import React from 'react';
import Title from './Title';
import Search from './Search';

export const Header = () => (
        <header className="header-section">
            <Title />
            <Search />
        </header>
);

export default Header;
