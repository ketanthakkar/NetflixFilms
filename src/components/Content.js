import React, { Component } from 'react';
import Filter from './Filter';
import MovieItem from './MovieItem';
import ErrorBoundary from '../error/ErrorBoundary';
import Header from './Header';
import Footer from './Footer';
import { connect } from 'react-redux';
import NoMovieFound from './NoMovieFound';
import PropTypes from 'prop-types';
import { fetchSearchedMovies } from '../actions/index'

class Content extends Component {

    componentDidMount() {
        
        if(this.props.match) {
            this.props.fetchSearchedMovies(this.props.match.params.query);
        }
    }

    render() {
        const { movies, showSearch } = this.props;
        
        return (
            <React.Fragment>
                {showSearch && <Header /> } 
                <ErrorBoundary>
                    {movies.length > 0 &&
                        <React.Fragment>
                            <Filter />
                            <main className="moviedata-container"> {
                                movies.map(movie => <MovieItem key={movie.id} movieItem={movie} />)
                            }
                            </main>
                        </React.Fragment>
                    }
                    { movies.length <= 0 && <NoMovieFound /> }
                </ErrorBoundary>    
                {showSearch && <Footer /> }
            </React.Fragment>
        );
    }
}

Content.propTypes = {
    showSearch: PropTypes.bool
};

Content.defaultProps = {
    showSearch: true
};

const mapDispatchToProps = { fetchSearchedMovies };

const mapStateToProps = (state) => ({
    movies: state.movies.movieData,
});

export { Content }; 
export default connect(mapStateToProps, mapDispatchToProps)(Content);