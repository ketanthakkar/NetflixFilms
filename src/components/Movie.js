import React, { Component } from 'react';
import Title from './Title';
import Footer from './Footer';
import Content from './Content';
import { connect } from 'react-redux';

export class Movie extends Component {

    render() {
        const { movie } = this.props;

        return (
            <article className="movie-container">
                <header className="header-section">
                    <Title />
                </header>
                <section className="movie-poster">
                    <img className="movie-img" src={movie.poster_path} alt={movie.tagline} />
                    <div className="movie-detail">
                        <span className="movie-title">{movie.title.toUpperCase()}</span>
                        <span className="movie-genres">{movie.genres.join(" & ")}</span>
                        <div>
                            <span className="movie-year">{movie.release_date}</span>
                            <span className="movie-length">{`${movie.runtime} min`}</span>
                        </div>
                        <p className="movie-description">{movie.overview}</p>
                    </div>    
                </section>
                <Content movies={ movie } nosearch="nosearch" />
                <Footer />
            </article>
        )
    }
}

const mapStateToProps = (state, pageProps) => ({
    movie: state.movies.movieData.find(movieData => movieData.id === parseInt(pageProps.match.params.id))
});

export default connect(mapStateToProps, null)(Movie);