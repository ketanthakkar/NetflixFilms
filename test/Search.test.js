import React from 'react';
import { mount } from 'enzyme';
import { Search as SearchComponent } from "../src/components/Search";
import { BrowserRouter as Router } from 'react-router-dom';

describe('should render Search correctly', () => {
  const tree = mount(
      <Router>
        <SearchComponent />
      </Router>
  );
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });
});
