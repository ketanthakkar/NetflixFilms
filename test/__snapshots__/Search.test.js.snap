// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Search correctly Snapshot test 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Search
      onSearch={[Function]}
    >
      <section
        className="search-section"
      >
        <h4
          className="white-text"
        >
          FIND YOUR MOVIE
        </h4>
        <input
          className="white-text search-title"
          onChange={[Function]}
          placeholder="Enter title/genre"
          type="text"
        />
        <div
          className="search-container"
        >
          <div
            className="search-selection"
          >
            <label
              className="white-text"
            >
              SEARCH BY
            </label>
            <button
              className="white-text nonselected-color"
              id="title-btn"
              onClick={[Function]}
            >
              TITLE
            </button>
            <button
              className="white-text nonselected-color"
              id="genre-btn"
              onClick={[Function]}
            >
              GENRE
            </button>
          </div>
          <Link
            replace={false}
            to="/search/Search undefined"
          >
            <a
              href="/search/Search undefined"
              onClick={[Function]}
            >
              <button
                className="white-text"
              >
                SEARCH
              </button>
            </a>
          </Link>
        </div>
      </section>
    </Search>
  </Router>
</BrowserRouter>
`;
